src/GeneratedHydrator/Configuration.php

```diff
@@ -164,7 +164,7 @@ public function getGeneratedClassesNamespace()
      */
     public function setGeneratedClassesTargetDir($generatedClassesTargetDir)
     {
-        $this->generatedClassesTargetDir = (string) $generatedClassesTargetDir;
+        $this->generatedClassesTargetDir = $generatedClassesTargetDir;
```

Comment:
This breaks internal data integrity of the `Configuration` object, and may cause exceptions at a later execution point in time.